etna/core/models/mixins.py

```diff
@@ -25,3 +29,47 @@ class ContentWarningMixin(models.Model):
 
     class Meta:
         abstract = True
+
+
+class NewLabelMixin(models.Model):
+    """Mixin to allow editors to toggle 'new' label to be applied on-publish"""
+
+    mark_new_on_next_publish = models.BooleanField(
+        verbose_name="mark this page as 'new' when published",
+        default=True,
+        help_text="Mark this as true before publishing, to display the 'new label'",
+    )
+
+    newly_published_at = models.DateField(
+        editable=False,
+        default=None,
+        null=True,
+    )
+
+    new_label_end_date = 21
+
+    def with_content_json(self, content):
+        obj = super().with_content_json(content)
+        if obj.mark_new_on_next_publish:
+            obj.mark_new_on_next_publish = False
+            obj.newly_published_at = datetime.now().date()
+        return obj
+
+    @cached_property
+    def is_new(self):
```

Comment:
I know I suggested this in the first place, but maybe `is_newly_published` would be a better fit? `is_new` sounds a little bit too generic (could potentially clash with something internal to Django/Wagtail somewhere), and isn't as descriptive as it could be.